src/rpc/blockchain.cpp

```diff
@@ -1597,6 +1599,236 @@ UniValue savemempool(const JSONRPCRequest& request)
+        m_could_reserve = true;
+        return true;
+    }
+
+    ~CoinsViewScanReserver() {
+        std::lock_guard<std::mutex> lock(g_utxosetscan);
+        if (m_could_reserve) {
+            g_scan_in_progress = false;
+        }
+    }
+};
+
+UniValue scantxoutset(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() < 1 || request.params.size() > 2)
+        throw std::runtime_error(
+            "scantxoutset <action> {\"pubkeys\": [\"pubkey\",...], \"xpubs\":[{\"xpub\": \"<xpub>\"}], other options}\n"
+            "\nScans the unspent transaction output set for possible entries that matches common scripts of given public keys.\n"
+            "\nArguments:\n"
+            "1. \"action\"                      (string, required) The action to execute\n"
+            "                                          \"start\" for starting a scan\n"
+            "                                          \"abort\" for aborting the current scan\n"
+            "                                          \"status\" for progress report (in %) of the current scan\n"
+            "2. \"options\"                     (object, optional)\n"
+            "      \"pubkeys\":[\"pubkey\",...]   (array of strings, optional) An array of HEX encoded public keys\n"
+            "      \"xpubs\":                   (array of xpub objects that will be used to derive child keys with the given lookup window after m/0/k and m/1/k scheme)\n"
+            "           [\n"
+            "               {\"xpub\":\"<xpub>\", (base58check encoded extended public key (xpub)\n"
+            "                \"lookupwindow\": [<startindex>, <stopindex>] (An array with two integers that does define the range of keys that will be deriven for the given xpubs, default is 0 to 1000)\n"
+            "                }\n"
+            "           ]\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"unspents\": [\n"
+            "    {\n"
+            "    \"txid\" : \"transactionid\",    (string) The transaction id\n"
+            "    \"vout\": n,                   (numeric) the vout value\n"
+            "    \"scriptPubKey\" : \"script\",   (string) the script key\n"
+            "    \"amount\" : x.xxx,            (numeric) The total amount in " + CURRENCY_UNIT + " received by the address\n"
+            "    \"height\" : n,                (numeric) Height of the unspent transaction output\n"
+            "   }\n"
+            "   ,...], \n"
+            " \"total_amount\" : x.xxx,         (numeric) The total amount of all found unspent outputs in " + CURRENCY_UNIT + "\n"
+            "]\n"
+        );
+
+    RPCTypeCheck(request.params, {UniValue::VSTR, UniValue::VOBJ});
+
+    UniValue result(UniValue::VOBJ);
+    if (request.params[0].get_str() == "status") {
+        CoinsViewScanReserver reserver;
+        if (reserver.reserve()) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "No scan in progress");
```

Comment:
Missing test for error.

Reply:
You mean missing a functional test for that case. Yes. I though about it, but would require a mockup-slowdown argument (`-testslowdown` or similar). Otherwise I guess it's hard to properly test this.

Reply:
Yes, agree that `status` should not raise an error when no scan is in progress.